crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -51,6 +52,7 @@ pub(crate) mod tests;
 
 const TASKS_PER_CORE: usize = 1;
 const END_OF_EPOCH_BROADCAST_CHANNEL_CAPACITY: usize = 2;
+const LOCAL_EXECUTION_TIMEOUT: Duration = Duration::from_secs(10);
```

Comment:
This may be a bit too conservative? Maybe a placeholder of 1min? Hanging is also ok as long as we print digests of txns missing effects periodically.